File: Homework/Mobile Application Development/Midterm Project/app/src/main/java/studio/xmatrix/qqpvp/assistant/ui/activity/menu/MenuHandler.java
package studio.xmatrix.qqpvp.assistant.ui.activity.menu;

import android.view.View;

import studio.xmatrix.qqpvp.assistant.databinding.MenuActivityBinding;
import studio.xmatrix.qqpvp.assistant.ui.activity.collection.CollectionActivity;
import studio.xmatrix.qqpvp.assistant.ui.activity.list.ListActivity;
import studio.xmatrix.qqpvp.assistant.ui.activity.search.SearchActivity;

public class MenuHandler {
    private MenuActivity activity;
    private MenuActivityBinding binding;

    MenuHandler(MenuActivity activity, MenuActivityBinding binding) {
        this.activity = activity;
        this.binding = binding;
    }

    public void onClickList(View view) {
        ListActivity.startActivity(activity, view.getId());
    }

    public void onClickSearch(View view) {
        SearchActivity.startActivity(activity);
    }

    public void onClickCollection(View view) {
        CollectionActivity.startActivity(activity);
    }
}

File: Homework/Mobile Phone Application Development/Lab5 WebAPI/app/src/main/java/com/icytown/course/experimentfive/webapi/ui/menu/MenuHandler.kt
package com.icytown.course.experimentfive.webapi.ui.menu

import com.icytown.course.experimentfive.webapi.databinding.MenuActivityBinding
import com.icytown.course.experimentfive.webapi.ui.bilibili.BilibiliActivity
import com.icytown.course.experimentfive.webapi.ui.github.GithubActivity

class MenuHandler(val activity: MenuActivity, val binding: MenuActivityBinding) {

    fun onClickBilibili() {
        BilibiliActivity.startActivity(activity)
    }

    fun onClickGithub() {
        GithubActivity.startActivity(activity)
    }
}
